docs/design/architecture-and-requirements.md

```diff
@@ -0,0 +1,351 @@
+
+For example, there're many machine learning platform built on top of K8s. It is relatively easy to get a K8s from a cloud vendor, easy to orchestrate machine learning required services/daemons run on K8s. However, K8s doesn't offer good support jobs like Spark/Flink/Hive. So if your company has Spark/Flink/Hive running on YARN, there're gaps and a significant amount of work to move required jobs from YARN to K8s. Maintaining a separate K8s cluster is also overhead to Hadoop-based data infrastructure.
+
+Similarly, if your company's data pipelines are mostly built on top of cloud resources and SaaS offerings. Asking you to install a separate YARN cluster to run a new machine learning platform doesn't make a lot of sense.
+
+### Pain \#3 Data scientist are forced to interact with lower-level platform components
+
+In addition to the above pain, we do see Data Scientists are forced to learn underlying platform knowledge to be able to build a real-world machine learning workflow.
+
+For most of the data scientists we talked with, they're experts of ML algorithms/libraries, feature engineering, etc. They're also most familiar with Python, R, and some of them understand Spark, Hive, etc. 
+
+If they're asked to do interactions with lower-level components like fine-tuning a Spark job's performance; or troubleshooting job failed to launch because of resource constraints; or write a K8s/YARN job spec and mount volumes, set networks properly. They will scratch their heads and typically cannot perform these operations efficiently.
+
+### Pain \#4 Comply with data security/governance requirements
+
+TODO: Add more details.
+
+### Pain \#5 No good way to reduce routine ML code development
+
+After the data is prepared, the data scientist needs to do several routine tasks to build the ML pipeline. To get a sense on the existing data set, it usually needs a split of the data set, the statistics of data set. These tasks have a common duplicate part of code which reduces the efficiency of data scientists.
+
+An abstraction layer/framework to help developer to boost ML pipeline development could be valuable. It's better that the developer only needs to fill callback function to focus on their key logics.
+
+# Submarine
+
+## Overview
+
+### A little bit history
+
+Initially, Submarine is built to solve problems of running deep learning jobs like Tensorflow/PyTorch on Apache Hadoop YARN, allows admin to monitor launched deep learning jobs, and manage generated models. 
+
+It was part of YARN initially, code resides under `hadoop-yarn-applications`. Later, the community decided to move to a subject of Hadoop because we want to support other resource management platforms like K8s. And finally, we're reconsidering Submarine's charter and Hadoop community voted that it is the time to moved Submarine to a separate Apache TLP.
```

Comment:
"move to a subject of Hadoop" --> not clear what exactly this means. "convert it to be a subproject within Hadoop"?

Reply:
Addressed.